fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/JDBCParameterBuildingVisitor.java

```diff
@@ -153,73 +205,88 @@ public boolean visit(java.lang.String elementName, int elementIndex, com.ibm.fhi
     @Override
     public boolean visit(String elementName, int elementIndex, com.ibm.fhir.model.type.String value) {
         Parameter p = new Parameter();
-        p.setName(name);
-        p.setValueString(value.getValue());
+        if (STRING.equals(searchParamType)) {
+            p.setValueString(value.getValue());
+        } else if (TOKEN.equals(searchParamType)) {
```

Comment:
in the case of the monitoring program, it'll store the value now in the setValueCode and be inserted as such.  

Does any existing parameters need to be updated? I think they are registered in a parameters table, and drives the query through to the TOKEN table for the Resource type.  in this case, it may be OK as there are no running instances, but may need to be carefully considered on how to update the parameter relationship to point to the right table

Reply:
I think not.  The changes how we're extracting the values, but shouldn't affect the actual parameter names or types at all.  The query was always looking in the X_TOKEN_VALUES table...but now we will actually store the value there :-)